packages/block-library/src/navigation/style.scss

```diff
@@ -51,25 +51,27 @@ $navigation-icon-size: 24px;
 		padding: 0;
 	}
 
-	// Force links to inherit text decoration applied to navigation block.
-	// The extra selector adds specificity to ensure it works when user-set.
-	&[style*="text-decoration"] {
-		.wp-block-navigation-item,
-		.wp-block-navigation__submenu-container {
-			text-decoration: inherit;
+	// The following rules provide class based application of user selected text
+	// decoration via block supports.
+	&.has-text-decoration-underline .wp-block-navigation-item__content {
+		text-decoration: underline;
+
+		&:focus,
+		&:active {
+			text-decoration: underline;
 		}
+	}
 
-		a {
-			text-decoration: inherit;
+	&.has-text-decoration-line-through .wp-block-navigation-item__content {
+		text-decoration: line-through;
 
-			&:focus,
-			&:active {
-				text-decoration: inherit;
-			}
+		&:focus,
+		&:active {
+			text-decoration: line-through;
 		}
 	}
 
-	&:not([style*="text-decoration"]) {
+	&:not([class*="has-text-decoration"]) {
```

Comment:
I wonder if it would be worth wrapping the `:not` in a `:where` to reduce specificity, since the only thing we're meant to be overriding here is browser defaults.

Reply:
👍 Thanks. I've added the use of `:where` in https://github.com/WordPress/gutenberg/pull/36345/commits/63516f69ff699d86f3376f0e3a7be73ca3b7c8e6